### ISSUE TITLE
Incorrect background on Lollipop for a GIF displayed asBitmap()

### ISSUE THREAD

**Issue by lynfogeek** [bumptech/glide#423]
Let's say I want to load a static version of this GIF into an `ImageView`: 

![miaow](https://media1.giphy.com/media/i9BL9JvYycxoI/200.gif)

I am currently using this:
```
Glide.with(getContext())
            .load(image.getUrl())
            .asBitmap()
            .diskCacheStrategy(DiskCacheStrategy.SOURCE)
            .placeholder(R.drawable.ic_placeholder)
            .into(imageView);
```

As a result, the image is properly loaded on a device / emulator pre-lollipop (with transparent background) but it will be drawn with a white background on Lollipop.

Some other gifs will even be displayed with a purple, blue or black background. [Like this one](http://giphy.com/gifs/hello-kitty-gif-Z4aSd8FdqMA6Y):
![image](https://cloud.githubusercontent.com/assets/1833455/7215397/44792106-e5d6-11e4-83ec-6cfaa2504693.png)

*note*: If i load them `.asGif()` the background will be transparent as expected

Is there any workaround to get a transparent background on Lollipop?

**Comment by sjudd** [comment]
There's a couple of potential causes here. Images with black backgrounds may be caused by an incorrect Bitmap configuration, you can test by either switching to [ARGB_8888 globally][1], or by using the [``format()``][2] method.

Another potential problem is that when we re-use Bitmaps, we're somehow failing to clear out the background of the image. You can test this by passing in [``BitmapPoolAdapter``][3] to the [``GlideBuilder`` class Glide uses for configuration][4] to temporarily disable Bitmap re-use and see if that fixes the problem.

Finally, it's also possible that Android's [``BitmapFactory``][5] class is decoding some of these images with backgrounds whereas our [``GifDecoder``][6] class is not. ``asGif()`` will force the image to be decoded using our ``GifDecoder``, ``asBitmap()`` will force the image to be decoded as a single frame using ``BitmapFactory``, and using neither ``asBitmap()`` nor ``asGif()`` will decode GIFs with more than one frame using our ``GifDecoder`` and single frame GIFs with ``BitmapFactory``.

tldr: try forcing ARGB_8888, then try disabling Bitmap reuse via BitmapPoolAdapter, then try just decoding the images with [``BitmapFactory#decodeStream``][7] directly.

If you can let me know what happens after each of those steps and/or post urls to the original images so I can (eventually) get to testing them myself, I'd bet we can make progress. Thanks for the report!

[1]: https://github.com/bumptech/glide/wiki/Configuration#bitmap-format
[2]: http://bumptech.github.io/glide/javadocs/350/com/bumptech/glide/BitmapRequestBuilder.html#format(com.bumptech.glide.load.DecodeFormat) 
[3]: http://bumptech.github.io/glide/javadocs/350/com/bumptech/glide/load/engine/bitmap_recycle/BitmapPoolAdapter.html
[4]: https://github.com/bumptech/glide/wiki/Configuration#bitmap-pool
[5]: http://developer.android.com/reference/android/graphics/BitmapFactory.html
[6]: http://bumptech.github.io/glide/javadocs/350/com/bumptech/glide/gifdecoder/GifDecoder.html
[7]: http://developer.android.com/reference/android/graphics/BitmapFactory.html#decodeStream(java.io.InputStream)

**Comment by lynfogeek** [comment]
Thank you @sjudd for you help. I tried these different solutions.

- Forcing ARGB_8888 via format() or globally doesn't not help
- Disabling the Bitmap reuse does not seem to help. I also tried to only load a single image, so the library should not need to reuse the bitmap. 

I am not sure to understand how can I decode it with `BitmapFactory#decodeStream`. But I assume the problem comes from the Android `BitmapFactory` because my problem only happen on Lollipop and works fine on KitKat / Jelly Bean.

Example of urls:
[White background instead of transparent](https://media4.giphy.com/media/i9BL9JvYycxoI/200.gif)
[Blue background instead of transparent](https://media0.giphy.com/media/Z4aSd8FdqMA6Y/200.gif)

**Comment by TWiStErRob** [comment]
@lynfogeek what he meant is to skip Glide and just manually load the image in `onCreate` for example:
```
imageView.setImageBitmap(BitmapFactory.decodeStream(...));
```
If you have wrong backgrounds this way, then it's a framework/gif file bug.

**Comment by TWiStErRob** [comment]
When you say Lollipop, is it 5.0, 5.0.x or 5.1?

**Comment by lynfogeek** [comment]
@TWiStErRob thank you. The background is wrong using `BitmapFactory#decodeStream` on Android Kitkat and Lollipop but it works fine with older versions.

Tested with more versions to be more precise:
* :white_check_mark: 4.1.1
* :white_check_mark: 4.3
* :x: 4.4.4
* :x: 5.0
* :x: 5.1

**Comment by sjudd** [comment]
@lynfogeek It does indeed look like an issue with BitmapFactory. If you can attach a minimal sample app that you used to test those various platform versions, I'll do my best to pass it along.

For now, you can use ``asGif()`` as a workaround. If you don't want the images to be animated, you can extend [``GlideDrawableImageViewTarget``][1], override ``onResourceReady()``, and always call ``stop()`` on the drawable after calling super. You could also look in to using a ``ResourceTranscoder`` to transform the GifDrawable into a simple Bitmap.

[1]: https://github.com/bumptech/glide/blob/master/library/src/main/java/com/bumptech/glide/request/target/GlideDrawableImageViewTarget.java

**Comment by sjudd** [comment]
Actually don't worry about the sample app, I'll look in to it.

**Comment by sjudd** [comment]
Internal bug is b/20567643

**Comment by sjudd** [comment]
I'm going to close this for now since I don't think we want to go to the effort (and take the performance hit) of working around the issue. 

If it seriously affects you, you can work around the problem yourself using asGif() to force the image to be decoded with Glide's GifDecoder.

**Comment by bearprada** [comment]
i got the same issue in SDK `v3.6.0`

**Comment by TWiStErRob** [comment]
Try working around it by https://github.com/bumptech/glide/issues/423#issuecomment-96232571 (second paragraph).

**Comment by tjych** [comment]
I HAVE THE SAME ISSUE. Glide load image，There is a blue background.

**Comment by TWiStErRob** [comment]
@tjych Did you try second paragraph of https://github.com/bumptech/glide/issues/423#issuecomment-96232571?

**Comment by tjych** [comment]
I load image , not gif , There is a blue background, My English is not good .

**Comment by anupamchugh** [comment]
If the format is PNG the black background won't be displayed.
